server/models.py

```diff
@@ -137,6 +136,17 @@ def __repr__(self):
     def by_name(name):
         return Course.query.filter_by(offering=name).one_or_none()
 
+    @property 
+    def display_name_with_semester(self):
+        year = self.offering[-2:len(self.offering)]
```

Comment:
I think it'll be safer to split on `/` characters since those are required in the offering tag.